packages/components/select/src/useSelect.ts

```diff
@@ -133,6 +135,12 @@ export const useSelect = (props, states: States, ctx) => {
     )
   )
 
+  const needStatusIcon = computed(() => form?.statusIcon ?? false)
+  const validateState = computed(() => formItem?.validateState || '')
+  const validateIcon = computed(
+    () => validateState.value && ValidateComponentsMap[validateState.value]
+  )
+
```

Comment:
Maybe it's possible to combine all three into one computed value.

Reply:
Do you mean something like this?
```
  const selectTagsStyle = computed(() => {
    const isHaveStatusIcon =
      form?.statusIcon &&
      formItem?.validateState &&
      validateState.value &&
      ValidateComponentsMap[validateState.value]
    return {
      maxWidth: `${
        unref(states.inputWidth) - 32 - (isHaveStatusIcon ? 22 : 0)
      }px`,
      width: '100%',
    }
  })
```

Reply:
Maybe as long as:
```ts suggestion
const showStatusIconAndState = computed(() => {
  return form?.statusIcon &&
      formItem?.validateState &&
      validateState.value &&
      ValidateComponentsMap[validateState.value]
})
```

Reply:
Hello, have any result of discussion? Anyway, I change the code from your guide so that this issue can be solved before the new version published.@btea

Reply:
I think keeping your latest changes is fine for now.

Reply:
Is it for `fix branch only do fix`? Actually, I'm targeting in `tag computed style` based on this branch(https://github.com/element-plus/element-plus/compare/dev...init-qy:element-plus:refactor-select), too many changes will trouble me.

Reply:
well,  let's keep it as it is.